atlasdb-api/src/main/java/com/palantir/atlasdb/metrics/MetricPublicationArbiter.java

```diff
@@ -50,12 +59,14 @@ public boolean test(MetricName metricName) {
     }
 
     public void registerMetricsFilter(MetricName metricName, MetricPublicationFilter filter) {
-        singleMetricFilters.merge(metricName, ImmutableList.of(filter), (oldFilters, newFilter)
-                -> ImmutableList.<MetricPublicationFilter>builder().addAll(oldFilters).addAll(newFilter).build());
+        singleMetricFilters.merge(metricName,
+                Collections.singleton(new DeduplicatingFilterHolder(filter)),
```

Comment:
nit: If you're using `ImmutableSet` elsewhere, the `ImmutableSet` javadoc recommends: 
```
Returns an immutable set containing {@code element}. 
Preferred over {@link Collections#singleton} for code consistency, 
{@code null} rejection, and because the return type conveys the immutability guarantee.
```